static/js/tippyjs.js

```diff
@@ -245,4 +245,28 @@ export function initialize() {
         },
         appendTo: () => document.body,
     });
+
+    delegate("body", {
+        target: ".message_inline_image > a > img",
+        appendTo: () => document.body,
+        // Add a short delay so the user can mouseover several inline images without
+        // tooltips showing and hiding rapidly
+        delay: [300, 20],
+        onShow(instance) {
+            // Some message_inline_images aren't actually images with a title,
+            // for example youtube videos, so we default to a generic title
+            this.title = $(instance.reference).parent().attr("title") || "attachment";
+            const prefix = $t({defaultMessage: "View or download"});
+            const suffix = " " + this.title;
+            const $markup = $("<span>").text(prefix).append($("<strong>").text(suffix));
+            instance.setContent($markup[0]);
+            // Prevent default browser tooltip
+            $(instance.reference).parent().removeAttr("title");
+        },
+        onHidden(instance) {
+            // Restore the original title attribute
+            $(instance.reference).parent().attr("title", this.title);
+            instance.destroy();
+        },
```

Comment:
I worry that this sort of logic moving around the `title` attribute risky being buggy. What do you think about just storing the content in an `aria-label` attribute (i.e. the label for screenreaders) and not the `title` attribute in HTML? It seems like it might let us remove the logic to live-edit the `title` attribute in the DOM.

Reply:
That's fair. The title attribute is set way up in the markdown processor. There are some various places in the frontend where it could be moved to a different attribute - but this did seem like the simplest way to do it at the time. I'll see if I can fine a clean way to remove the need to use the title attribute.

Reply:
Also just to be clear: the title attribute has to be removed (at least when there is a tippy tooltip showing) in order to prevent the browser default tooltip from showing.

Reply:
Yes, what I had in mind is having `clean_user_content_links` or `rendered_markdown.js` be responsible for any messing with the `title` attribute in a single pass before rendering to the user, rather than having it happen dynamically on hover.

Reply:
I removed the title logic from tippy, `title` is not replaced by `aria-label` in `clean_user_content_links`.